io.github.andrewmkiss.Demeter.yml

```diff
@@ -0,0 +1,190 @@
+app-id: io.github.andrewmkiss.Demeter
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: dathena
+finish-args:
+  - --socket=x11
+  - --share=ipc
+  - --filesystem=host
+
+modules:
+  - name: hdf4_lib
+    disabled: false 
+    buildsystem: simple
+    no-autogen: true
+    build-commands:
+      - 'hdf4_lib/install.sh'
+    sources:
+      - type: archive
+        url: https://support.hdfgroup.org/ftp/HDF/releases/HDF4.2.16/src/CMake-hdf-4.2.16.tar.gz
+        sha256: e8115564daa913c274f88558c2bc5d6685ef51531c80bf95f6dd92ced1f7b64f
+      - commands:
+        - set -e
+        - cd CMake-hdf-4.2.16/
+        - sed -i "79s/set.*/set \( INSTALLDIR \"\/app\")/" HDF4config.cmake
+        - ctest -S HDF4config.cmake,BUILD_GENERATOR=Unix -C Release -V -O hdf4.log
+        - cd build
+        - make install
+        - make clean
+        dest: hdf4_lib
+        dest-filename: install.sh
+        type: script
+
+  - name: gnuplot
+    disabled: false
+    buildsystem: simple
+    build-commands:
+      - ./configure --prefix=/app/ --without-qt
+      - make
+      - make install
+      - make clean
+    sources:
+      - type: archive
+        url: https://sourceforge.net/projects/gnuplot/files/gnuplot/6.0.0/gnuplot-6.0.0.tar.gz
+        sha256: 635a28f0993f6ab0d1179e072ad39b8139d07f51237f841d93c6c2ff4b1758ec
+
+  - name: ncurses
+    disabled: false
+    buildsystem: autotools
+    no-autogen: true
+    config-opts: --prefix=/usr
+    build-options:
+      cflags: '-fPIC'
+      cppflags: '-fPIC'
+      ldflags: '-fpic'
+    sources:
+      - type: git
+        url: https://github.com/mirror/ncurses.git
+        commit: 79b9071f2be20a24c7be031655a5638f6032f29f
+
+  - name: ifeffit
+    disabled: false
+    buildsystem: simple
+    build-commands:
+      - FFLAGS="-g -O2 -fPIC" ./configure --prefix=/app/ --without-pgplot
+      - sed -i "s/TERMCAP_LIB =.*/TERMCAP_LIB = -L\/app\/lib -lncurses/" src/cmdline/Makefile
+      - make
+      - make install
+      - sed -i "s/\/run\/build.*/\/app\/lib\/libnopgplot.a/" /app/share/ifeffit/config/Config.mak
```

Comment:
not sure why you do these obscure sed commands.

Reply:
Agreed, but they are required to build. I made the ncurses module more specific, e.g. grab a particular git tag instead of the `stable.tar.gz` file, and tested without these (failed in downstream modules)